models/DataObject/ClassDefinition/CustomLayout/Dao.php

```diff
@@ -70,7 +70,7 @@ public function getIdByName($name = null)
      */
     public function getNameById($id = null)
     {
-        $name = $this->db->fetchOne('SELECT name FROM custom_layouts WHERE id = ?', $id);
+        $name = $this->db->fetchOne('SELECT name FROM custom_layouts WHERE id = ?', (int) $id);
```

Comment:
id can be alphanumeric..so this fix will not work!

Reply:
how about a check here https://github.com/pimcore/pimcore/blob/2904c99cbc9fe9fad71069ac6e2bbf325552ddbd/models/DataObject/ClassDefinition/CustomLayout/Dao.php#L73 to fetch name only if "Id" is not null..? and then also remove the default value null

Reply:
Better approach:

```php
    /**
     * @param mixed $id
     *
     * @return string
     */
    public function getNameById($id)
    {
        try{
            $name = $this->db->fetchOne('SELECT name FROM custom_layouts WHERE id = ?', $id);
        } catch(\Exception $e) {
            $name = null;
        }

        return $name;
    }
```
and it will be great if can also change getIdByName() just like this. Thank you!

Reply:
You are right.

When I look into models/DataObject/ClassDefinition/Dao.php there are also two methods getNameById($id = null) and getIdByName($name = null) which do the same, but the errors are caught by a try/catch block. It does work, because name is always a scalar (string) and never null in this case.
```
try {
            $class = new self();
            $id = $class->getDao()->getIdByName($name);
            if ($id) {
                return self::getById($id);
            }
        } catch (\Exception $e) {
            return null;
        }
```
Adding a is_null check and changing docblock might be not consistent if done only in one class.
```
    /**
     * @param null $name
     *
     * @return int|null
     */
    public function getIdByName($name = null)
    {
        if (is_null($name)) {
            return null;
        }
        $name = $this->db->fetchOne('SELECT id FROM custom_layouts WHERE name = ?', $name);

        return $name;
    }

    /**
     * @param null $id
     *
     * @return string|null
     */
    public function getNameById($id = null)
    {
        if (is_null($id)) {
            return null;
        }
        $name = $this->db->fetchOne('SELECT name FROM custom_layouts WHERE id = ?', $id);

        return $name;
    }
```

Adding a try/catch block is kind of special in this case, because it's a \TypeError
```
    /**
     * @return bool
     */
    public function exists()
    {
        try {
            $name = $this->getDao()->getNameById($this->getId());
            return is_string($name);
        } catch (\Exception | \Error $e) {
        }

        return false;
    }
```
So the simplest solution would be to change \Pimcore\Model\DataObject\ClassDefinition\CustomLayout::exists() but this does not fix usage of non scalar/null values in getNameById() and getIdByName() in both Dao classes.
```
    /**
     * @return bool
     */
    public function exists()
    {
        if (is_null($this->getId())) {
            return false;
        }
        $name = $this->getDao()->getNameById($this->getId());

        return is_string($name);
    }
```

Please decide :)

Reply:
I think better would be to fix `getNameById` and `getIdByName`  in both classes `models/DataObject/ClassDefinition/Dao.php` & `models/DataObject/ClassDefinition/CustomLayout/Dao.php` by adding `is_null` check as it does not makes sense to search for rows with null id. thanks!

Reply:
See 2nd commit. Hope this is OK
Thanks